src/sentry/api/endpoints/project_ownership.py

```diff
@@ -14,6 +16,17 @@ class ProjectOwnershipSerializer(serializers.Serializer):
     fallthrough = serializers.BooleanField()
     autoAssignment = serializers.BooleanField()
 
+    @staticmethod
+    def _validate_no_codeowners(rules: List[Rule]):
+        """codeowner matcher types cannot be added via ProjectOwnership, only via github"""
```

Comment:
might just be me catching up here, but you can still add a codeowner file via the API directly right? which would just be using the `ProjectCodeOwners` table instead. 

`only via github` is a little confusing to me because we have gitlab too (supposedly) and i _thought_ manually as well